app/main.py

```
@@ -1,14 +1,67 @@
-class Validator:
-    pass
+from __future__ import annotations
+from abc import abstractmethod, ABC
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
 
+    def __get__(self, instance: BurgerRecipe, owner: type) -> int | str:
+        return getattr(instance, self.protected_name)
 
-class OneOf:
-    pass
+    def __set__(self, instance: BurgerRecipe, value: int | str) -> None:
+        self.validate(value)
+        setattr(instance, self.protected_name, value)
+
+    @abstractmethod
+    def validate(self, value: int | str) -> int | str:
```

The return type hint `-> int | str` is inconsistent with the method's purpose and the implementations in the child classes (`Number`, `OneOf`). The `validate` method should either raise an error or complete successfully, returning `None`. Consider changing the type hint to `-> None` to accurately reflect this.